src/libraries/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/DebugDirectory/DebugDirectoryBuilder.cs

```diff
@@ -114,7 +114,12 @@ public void AddEntry<TData>(DebugDirectoryEntryType type, uint version, uint sta
             }
 
             // We allow NUL characters to allow for padding for backward compat purposes.
-            if (pdbPath.Length == 0 || pdbPath.IndexOf('\0') == 0)
+            if (pdbPath.Length == 0 ||
+#if NETCOREAPP
+                pdbPath.StartsWith('\0'))
+#else
+                pdbPath[0] == '\0')
+#endif
```

Comment:
Why not use `pdbPath[0] == '\0'` unconditionally?

Reply:
Yeah, probably that is straight forward